### PR TITLE
Add separate `code-studio` bundle for Makerlab dependencies. Add Makerlab mode for Applab.

### PR DESCRIPTION
**Pull request by bcjordan**

### CODE REVIEW

**Comment by bcjordan** [comment]
New Rails test seems to fail in Travis, will investigate.

File: apps/package.json [hunk 1]
```diff
@@ -27,7 +27,10 @@
   },
   "browserify-global-shim": {
     "react": "React",
-    "react-dom": "ReactDOM"
+    "react-dom": "ReactDOM",
+    "johnny-five": "JohnnyFive",
+    "playground-io": "PlaygroundIO",
+    "chrome-serialport": "ChromeSerialport"
```
Comment (Bjvanminnen) [inline apps/package.json]: I suspect these better belong in browserify-shim rather than as global-shims. The only time we should need to have packages in global shim is if we're requiring other packages that require them (i.e. we require a color-picker module that requires React, thus it needs to be in global).

Not sure there's any harm in this, but probably better off putting these in browserify-shim.

Comment (Bjvanminnen) [inline apps/package.json]: You can see an example in https://github.com/code-dot-org/code-dot-org/pull/7583 where I do this for marked.

Comment (Bjvanminnen) [inline apps/package.json]: Are these actually used in apps at all yet? (I dont see any other changes to apps in this PR)

Comment (bcjordan) [inline apps/package.json]: Nope, have usage test locally but will split that out into a separate PR.

**Comment by Bjvanminnen** [comment]
lgtm. I might not bother with the changes to apps/package.json until they're needed, but if that PR is coming shortly this is fine.
